modules/ozoneBidAdapter.js

```diff
@@ -502,15 +617,99 @@ export const spec = {
       ret[tmp[0]] = tmp[1];
     }
     return ret;
+  },
+  /**
+   * This will be called IF we want to enforce gdpr on the client
+   * Do we have to block this request? Could be due to config values & gdpr permissions etc
+   * @return {boolean|*[]} true = block the request, else false
+   */
+  blockTheRequest(bidderRequest) {
+    // if there is an ozone.oz_request = false then quit now.
+    let ozRequest = config.getConfig('ozone.oz_request');
+    if (typeof ozRequest == 'boolean' && !ozRequest) {
+      utils.logError('OZONE: Will not allow auction : ozone.oz_request is set to false');
+      return true;
+    }
+    // is there ozone.oz_enforceGdpr == true (ANYTHING else means don't enforce GDPR))
+    let ozEnforce = config.getConfig('ozone.oz_enforceGdpr');
+    if (typeof ozEnforce != 'boolean' || !ozEnforce) { // ozEnforce is false by default
+      utils.logError('OZONE: Will not validate GDPR on the client : oz_enforceGdpr is not set to true');
+      return false;
+    }
+    //
+    // FROM HERE ON : WE ARE DOING GDPR CHECKS
+    //
+    // maybe the module was built without consentManagement module so we won't find any gdpr information
+    if (!bidderRequest.hasOwnProperty('gdprConsent')) {
+      return false;
+    }
+    // If there is indeterminate GDPR (gdprConsent.consentString == undefined or not a string), we will DITCH this:
+    if (typeof bidderRequest.gdprConsent.consentString !== 'string') {
+      utils.logError('OZONE: Will block the request - bidderRequest.gdprConsent.consentString is not a string');
+      return true;
+    }
+    // IF the consentManagement module sends through the CMP information and user has refused all permissions:
+    if (this.failsGdprCheck(bidderRequest)) {
+      return true;
+    }
+    return false;
+  },
+  /**
+   * Examine the gdpr information inside the bidderRequest and return the boolean answer to the question
+   * @param bidderRequest
+   * @return {boolean}
+   */
+  failsGdprCheck(bidderRequest) {
+    // maybe the module was built without consentManagement module
+    if (!bidderRequest.hasOwnProperty('gdprConsent')) {
```

Comment:
This check is being done outside of this function on line 643. Is it needed here, too?

Reply:
you are right - its not needed - submitting an updated PR shortly - thanks for flagging/bringing to our attention @msm0504